homeassistant/components/pvpc_hourly_pricing/sensor.py

```diff
@@ -0,0 +1,195 @@
+"""
+Sensor to collect the reference daily prices of electricity ('PVPC') in Spain.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/integrations/pvpc_hourly_pricing/
+"""
+from datetime import timedelta
+import logging
+from random import randint
+from typing import Optional
+
+from aiopvpc import PVPCData
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_NAME, CONF_TIMEOUT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.event import (
+    async_track_point_in_time,
+    async_track_time_change,
+)
+from homeassistant.helpers.restore_state import RestoreEntity
+import homeassistant.util.dt as dt_util
+
+from .const import ATTR_TARIFF, DEFAULT_TARIFF, DEFAULT_TIMEOUT
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_PRICE = "price"
+ICON = "mdi:currency-eur"
+UNIT = "€/kWh"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: config_entries.ConfigEntry, async_add_entities
+):
+    """Set up the electricity price sensor from config_entry."""
+    name = config_entry.data[CONF_NAME]
+    pvpc_data_handler = PVPCData(
+        tariff=config_entry.options.get(
```

Comment:
The tariff needs to be stored in data instead of options for it to be the unique_id.    If you can change it, it isn't unique

Reply:
Oops, I forgot to remove that. `entry.options` is not used anymore